user/user_lego_sensor.h

```diff
@@ -16,9 +16,18 @@
 #include <lego.h>
 #include <lego_sensor_class.h>
 
+/* The USER_LEGO_SENSOR_TEXT_VALUE_SIZE can be a maximum of PAGE_SIZE-2
+ * to take into account a trailing newline (added later) and NULL. We are
+ * limiting it to much less to save memory for each user_lego_sensor 
+ * instance.
+ */
+
+#define USER_LEGO_SENSOR_TEXT_VALUE_SIZE (512-2)
```

Comment:
I think it would be better to leave out the `-2` here. The newline is not ever included in this buffer, so you don't need to save room for it. Also, a null terminator *is* included in the buffer, so basically, right now you are limiting the buffer to 509 printable characters. 

Whatever you decide here should also be mentioned in the website documentation comments.